src/hooks/useIsVisible.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useRef, useState } from 'react'
+
+/** Determines if an element is partially visible in the viewport. */
```

Comment:
Should explain the purpose of the returned ref in the JSDOC.

Reply:
JSDoc Added: 37427088f917675d0fc014fa711d5212e4710c2a

Reply:
It appears `@param` comments do not work with VS Code's IntelliSense, but `/** ... */` comments do.

<img width="435" src="https://github.com/user-attachments/assets/20c87102-cc0d-4c80-9e9b-7d2b021d868c">

See: 9eef751896